diff --git a/arch/arm/boot/dts/imx6qdl-wandboard.dtsi b/arch/arm/boot/dts/imx6qdl-wandboard.dtsi
index 5fb0916..247ccd4 100644
--- a/arch/arm/boot/dts/imx6qdl-wandboard.dtsi
+++ b/arch/arm/boot/dts/imx6qdl-wandboard.dtsi
@@ -89,6 +89,28 @@
 	};
 };
 
+&i2c3 {
+  clock-frequency = <100000>;
+  pinctrl-names = "default";
+  pinctrl-0 = <&pinctrl_i2c3>;
+  status = "okay";
+
+	gyro: gyro@6a {
+		compatible = "st,lsm330dlc-gyro";
+		reg = <0x6a>;
+	};
+
+	accel: accel@18 {
+		compatible = "st,lsm330dlc-accel";
+		reg = <0x18>;
+	};
+
+	rtc: rtc@51 {
+		compatible = "nxp,pcf2127";
+		reg = <0x51>;
+	};
+};
+
 &iomuxc {
 	pinctrl-names = "default";
 
@@ -139,6 +161,13 @@
 			>;
 		};
 
+		pinctrl_i2c3: i2c3grp {
+			fsl,pins = <
+				MX6QDL_PAD_GPIO_5__I2C3_SCL			0x4001b8b1
+				MX6QDL_PAD_GPIO_16__I2C3_SDA		0x4001b8b1
+			>;
+		};
+
 		pinctrl_spdif: spdifgrp {
 			fsl,pins = <
 				MX6QDL_PAD_ENET_RXD0__SPDIF_OUT		0x1b0b0
@@ -247,16 +276,20 @@
 	status = "okay";
 };
 
-&usdhc1 {
+&usdhc1 {  /* Baseboard microSD slot */
 	pinctrl-names = "default";
 	pinctrl-0 = <&pinctrl_usdhc1>;
+	bus-width = <4>;
+	vmmc-supply = <&reg_3p3v>;
 	cd-gpios = <&gpio1 2 0>;
 	status = "okay";
 };
 
-&usdhc3 {
+&usdhc3 { /* Module microSD slot */
 	pinctrl-names = "default";
 	pinctrl-0 = <&pinctrl_usdhc3>;
+	vmmc-supply = <&reg_3p3v>;
 	cd-gpios = <&gpio3 9 0>;
 	status = "okay";
 };
+
